warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__lead_time.sql

```diff
@@ -0,0 +1,174 @@
+           LEAD (feed_key) OVER (PARTITION BY base64_url ORDER BY _valid_from ASC) AS next_feed_key,
+           LEAD (EXTRACT(date FROM _valid_from)) OVER (PARTITION BY base64_url ORDER BY _valid_from ASC) AS next_feed_valid_from,
+           EXTRACT(date FROM _valid_from) AS valid_from
+      FROM distinct_feed_versions
+),
+
+scheduled_trips_version_history AS (
+    SELECT t1.service_date,
+           t1.feed_key,
+           t2.trip_id,
+           t2.trip_stop_times_hash,
+           t2.trip_stop_locations_hash,
+           t3.base64_url,
+           t3.prev_feed_key,
+           t3.next_feed_key,
+           t3.next_feed_valid_from,
+           t3.valid_from
+      FROM all_scheduled_service t1
+      LEFT JOIN trips_expanded t2
+        ON t2.service_id = t1.service_id
+       AND t2.feed_key = t1.feed_key
+      JOIN feed_version_history t3
+        ON t3.feed_key = t1.feed_key
+    -- Since we are comparing feeds with their previous version, omit the initial version of every feed - no comparison is possible
+     WHERE t3.prev_feed_key IS NOT null
+),
+
+-- The self-outer-join, with all of the coalescing, allows us to see:
+------ 1 trips that were REMOVED since the previous version
+------ 2 trips that were ADDED since the previous version
+------ 3 trips that were CHANGED since the previous version
+trips_version_compare AS (
+  SELECT
+         -- base64_url is same between feed versions
+         COALESCE(trips.base64_url,prev_trips.base64_url) AS base64_url,
+         -- one feed's key is the previous feed's next key
+         COALESCE(trips.feed_key,prev_trips.next_feed_key) AS feed_key,
+         -- one feed's previous key is the previous feed's key
+         COALESCE(trips.prev_feed_key,prev_trips.feed_key) AS prev_feed_key,
+         -- we need to know the next feed's valid_from date, in cases where a trip is removed since the previous feed
+         COALESCE(trips.valid_from,prev_trips.next_feed_valid_from) AS valid_from,
+         DATE_DIFF(
+                     COALESCE(trips.service_date, prev_trips.service_date),
+                     COALESCE(trips.valid_from,prev_trips.next_feed_valid_from),
+                     DAY
+         ) AS days_until_service_date,
+         trips.trip_id,
+         trips.trip_stop_times_hash,
+         trips.trip_stop_locations_hash,
+         prev_trips.trip_id AS prev_trip_id,
+         prev_trips.trip_stop_times_hash AS prev_trip_stop_times_hash,
+         prev_trips.trip_stop_locations_hash AS prev_trip_stop_locations_hash
+    FROM scheduled_trips_version_history AS trips
+    FULL OUTER JOIN scheduled_trips_version_history AS prev_trips
+      ON trips.prev_feed_key = prev_trips.feed_key
+     AND trips.trip_id = prev_trips.trip_id
+     AND trips.service_date = prev_trips.service_date
+),
+
+-- We count each "infraction" type seperately, mostly for QA purposes
+----  Note that since all_scheduled_service table has a row per trip per service day,
```

Comment:
Massive nit pick but `all_scheduled_service` is purely service ID (not trip) level

Reply:
i appreciate it nonetheless! updated the comment to be a bit more vague